### ISSUE TITLE
pull deleting not just content, but also parent folder

### ISSUE THREAD

**Issue by mobileWMS** [odeke-em/drive#577]
Hello,
     I have a folder called "$HOME/intran/wmsappdrive", is what I run "init" against ( drive init $HOME/intran/wmsappdrive ).  The "drive init" works perfectly, and for hours "drive pull" works fine ( via a shell script ).  But every so often, a pull will delete not only all the content of "wmsappdrive", but also delete "wmsappdrive".  This just started happening.

As one example, from my logs:
1. Check for existence of "wmsappdrive" using normal bash commands - it existed as of 14:32:16.997948109
2.  Inside wmsappdrive, run `drive pull -no-prompt=true -verbose` command.
3.  After this pull, check for existence of "wmsappdrive" using normal bash commands, and find that wmsappdrive does not exist as of 14:33:30.389852364

Did something change in the most recent version that makes the above command not work anymore?

Here is the version information:
```
[wmsodbc]> drive version
drive version: 0.3.4
Commit Hash: '9ba58541a7a40675f389bdad3da303b6140a059c'
Go Version: go1.5.1
OS: linux/amd64
BuildTime: 2016-01-26 12:20:17.989524924 -0500 EST
```

And the OS information:
```
[wmsodbc]> lsb_release -a
LSB Version:    :base-4.0-amd64:base-4.0-noarch:core-4.0-amd64:core-4.0-noarch
Distributor ID: CentOS
Description:    CentOS release 6.7 (Final)
Release:        6.7
Codename:       Final
`

**Comment by odeke-em** [comment]
Hello @mobileWMS, thanks for the question and welcome to drive.
So just to let you know, drive works also with specific directories. You don't have to pull from the parent directory always. If you pull from the parent directory, that means all the files that are in sub-directories and aren't matched up with what is on remote will be removed as it seems in your case.
If you'd like to pull specific files, you can do so like this
```shell
$ cd wmsappdrive
$ drive pull myDir1 myDir2
```
instead of always
```shell
$ cd wmsappdrive
$ drive pull
```

Also there are non-destructive operations that you could toggle by using flags `--exclude-ops` or `--no-clobber`. Please see https://github.com/odeke-em/drive#pulling.

**Comment by mobileWMS** [comment]
Yes, I use the "drive pull myDir1" in other places.  Just in the old version, I did not have this issue, so was sort of wondering if something changed.  What you said definitely makes sense, so I cannot really complain if you did decide to change the behavior, it just surprised me.

All that said, I will see what happens with --exclude-ops and --no-clobber.

**Comment by matthewlenz** [comment]
This has happened to me 2 times now only it's completely random over the last couple months (I'm running 0.3.5).  I have a very simple script that work 99% of the time and then seemingly at random I'll get an error saying the directory no longer exists:

#!/bin/bash

if [ ! -d $1 ];
then
	echo "invalid directory $1"
	exit 1
fi

cd $1
drive pull -no-prompt=true -quiet=true >/dev/null
mv *.pdf *.tif $2/ 2>/dev/null
drive push -no-prompt=true -quiet=true >/dev/null
drive emptytrash -no-prompt=true -quiet=true >/dev/null

The above script always called in the exact same way:

google_drive_sync gdfolder destfolder

**Comment by n-st** [comment]
I'm seeing the same problem with 0.3.9.1 on FreeBSD: I'm running `drive` from cron and every few months, it just decides to delete the entire local structure, including the `.gd` and parent directory.  
Restoring the directory from backup fixes the problem for a few more months.

My crontab is invoking `drive` like this:

    drive pull -export pdf,txt,doc,odt,xls,ods,ppt,odp -fix-clashes -no-prompt -verbose /mnt/backups/googledrive/johndoe

When `drive` decides to wipe out its target, it produces the following output:

    Resolving...
    localDelete: "/mnt/backups/googledrive/johndoe" remove /mnt/backups/googledrive/johndoe: permission denied
    / err: remove /mnt/backups/googledrive/johndoe: permission denied  

I'm seeing absolutely no pattern to how/when this is happening. I have multiple crontab entries for multiple Google Drive users (on the same system, but using a mutex so only one `drive` instance can run at any time) and it seems to hit them completely randomly — usually only one account at a time, occasionally two in quick succession.

**Comment by odeke-em** [comment]
Hello @n-st. Am a little confused now from your command
>...drive pull -export pdf,txt,doc,odt,xls,ods,ppt,odp -fix-clashes -no-prompt -verbose /mnt/backups/googledrive/johndoe

I think you are pushing using the `-m` flag but then while pulling you assume that the paths are the same yet `/mnt/backups/googledrive/johndoe` is just a symlink. and it seems like your `/mnt/backups/googledrive/johndoe` is nested inside another Google Drive init'd directory i.e
```
|-where you are running drive currently
  |-/mnt/backups/*
```

**Comment by n-st** [comment]
Hi @odeke-em,

> I think you are pushing using the -m flag but then while pulling you assume that the paths are the same

I'm actually not pushing at all, only pulling.  
I'm not entirely sure how I ran `drive init` back in the day, but the only `.gd` directories are 
 at `/mnt/backups/googledrive/johndoe/.gd`, `/mnt/backups/googledrive/janesmith/.gd`, and so forth. `johndoe` etc are all regular directories; `/mnt/backups/googledrive` is a nullfs mount (the FreeBSD equivalent of `mount --bind`).  
`/mnt/backups/googledrive`, `/mnt/backups`, `/mnt`, `/`, and `$HOME` all do *not* contain a `.gd` directory.  
Also, `drive` complains about "no drive context found" on successive runs after the bug has occurred, so there don't seem to be any conflicting drive inits anywhere within reach.
